src/screens/MovieDetailScreen.tsx

```diff
@@ -1,16 +1,372 @@
+  },
+  cast_heading: {
+    color: COLORS.White,
+    fontSize: FONTSIZE.size_20,
+    fontWeight: '600',
+    marginVertical: SPACING.space_10,
+  },
+  container_gap: {
+    gap: SPACING.space_24,
+  },
+  cast_container: {
+    paddingHorizontal: SPACING.space_18,
+  },
+  btn: {
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  seat_btn: {
+    backgroundColor: COLORS.Orange,
+    paddingHorizontal: SPACING.space_24,
+    paddingVertical: SPACING.space_8,
+    borderRadius: BORDERRADIUS.radius_20,
+    marginVertical: SPACING.space_8,
+    width: 130,
+  },
+  seat_text: {
+    fontSize: FONTSIZE.size_12,
+    fontWeight: '500',
+    color: COLORS.White,
+    textAlign: 'center',
+  },
 });
 
-const MovieDetailScreen = (props: Props) => {
+const movie_details_fun = async (movie_id: any) => {
+  try {
+    let response = await axios.get(movie_details(movie_id));
+    return response.data;
+  } catch (error) {
+    console.log(' something went wrong in movie_details_fun ', error);
+  }
+};
+
+const movie_case_details_fun = async (movie_id: any) => {
+  try {
+    let response = await axios.get(movie_cast_details(movie_id));
+    return response.data;
+  } catch (error) {
+    console.log(' something went wrong in movie_case_details_fun ', error);
+  }
+};
+
+const MovieDetailScreen = ({navigation, route}: any) => {
+  const [is_loading, set_is_loading] = useState<boolean>(false);
+  const [movie_details, set_movie_details] = useState<any>(undefined);
+  const [movie_cast_details, set_movie_cast_details] = useState<any>(undefined);
+
+  const movie_id = route.params.movie_id;
+
+  const handle_goback = () => {
+    navigation.goBack();
+  };
+
+  useEffect(() => {
+    set_is_loading(true);
+    const get_movies_data = async () => {
+      const movie_details_data = await movie_details_fun(movie_id);
+      set_movie_details(movie_details_data);
+
+      const movie_cast_details_data = await movie_case_details_fun(movie_id);
+      set_movie_cast_details(movie_cast_details_data);
+    };
+
+    get_movies_data();
+
+    set_is_loading(false);
+  }, []);
+
+  const format_date = (input_date: string) => {
+    const [year, month, day] = _.split(input_date, '-');
+
+    const month_names = {
+      '01': 'January',
+      '02': 'February',
+      '03': 'March',
+      '04': 'April',
+      '05': 'May',
+      '06': 'June',
+      '07': 'July',
+      '08': 'August',
+      '09': 'September',
+      '10': 'October',
+      '11': 'November',
+      '12': 'December',
+    };
+
+    const formatted_date = `${day} ${month_names[month]} ${year}`;
+
+    return formatted_date;
+  };
+
+  const handle_seat_select = () => {
+    navigation.push('SeatBooking', {
+      background_image: base_image_path('w780', movie_details?.backdrop_path),
+      poster_image: base_image_path('original', movie_details?.poster_path),
+    });
+  };
+
+  if (is_loading) {
+    return (
+      <View style={styles.container}>
+        <StatusBar hidden />
```

Comment:
`StatusBar`, `Header` are rendered regardless of the is_loading condition. Prefer to impose condition only on the set of components that needs to be rendered dynamically.